Automattic/wp-calypso
# Refactor fetching OAuth2 client info, remove cachingActionCreatorFactory

Removes the `cachingActionCreatorFactory` abstraction in `state/utils` that was only ever used once and rewrites the `fetchOAuth2ClientData` code without it. The result is a very simple action thunk.

When using WP.com login to login into an OAuth app (i.e., the `client_id` query arg is present), `fetchOAuthClientData` fetches information about the OAuth app so that it can customize the login page. That happens both in SSR (the page.js route handler waits until the fetch is complete, and the results are cached using the `lru` package) and in client-side JS.

**How to test:**
Verify that when loading a SSR-ed login page from URL like this:
```
http://calypso.localhost:3000/log-in?client_id=930&redirect_to=https://wordpress.com/theme?client_id=930
```
then a customized VaultPress login page is returned from server:
<img width="468" alt="Screenshot 2019-10-08 at 12 13 24" src="https://user-images.githubusercontent.com/664258/66387869-2f208400-e9c5-11e9-8b59-b8b6f5705204.png">

Verify that the SSR-ed header for a custom OAuth app (one I registered at developer.wordpress.com) contains the app name:
```
http://calypso.localhost:3000/log-in?client_id=58883&redirect_to=https://wordpress.com/theme?client_id=58883
```
renders as:
<img width="468" alt="Screenshot 2019-10-08 at 12 14 03" src="https://user-images.githubusercontent.com/664258/66387897-419abd80-e9c5-11e9-80ef-786efe394f22.png">

(notice that I disabled JS in Chrome Devtools to make sure I see the SSR-ed page)

Verify that the same rendering also happens client-side.
